src/mqt/core/qiskit_utils/imports.py

```diff
@@ -0,0 +1,337 @@
+"""Functionality for interoperability with Qiskit."""
+from __future__ import annotations
+
+import re
+import sys
+from typing import TYPE_CHECKING, cast
+
+from qiskit.circuit import AncillaQubit, AncillaRegister, Clbit, Instruction, ParameterExpression, Qubit
+
+from mqt.core._core import (
+    Control,
+    Expression,
+    OpType,
+    QuantumComputation,
+    StandardOperation,
+    SymbolicOperation,
+    Term,
+    Variable,
+)
+
+if TYPE_CHECKING:
+    from qiskit import QuantumCircuit
+    from qiskit.transpiler import Layout
+
+
+def qiskit_to_mqt(qiskit_circuit: QuantumCircuit) -> QuantumComputation:
+    """Convert a Qiskit circuit to a QuantumComputation object.
+
+    Args:
+        qiskit_circuit (QuantumCircuit): The Qiskit circuit to convert.
+
+    Returns:
+        QuantumComputation: The converted circuit.
+    """
+    mqt_computation = QuantumComputation()
+
+    if qiskit_circuit.name is not None:
+        mqt_computation.name = qiskit_circuit.name
+
+    qubit_index = 0
+    qubit_map = {}
+    qiskit_qregs = qiskit_circuit.qregs
+    for reg in qiskit_qregs:
+        size = reg.size
+        name = reg.name
+        if isinstance(reg, AncillaRegister):
+            mqt_computation.add_ancillary_register(size, name)
+            for i in range(size):
+                qubit_map[AncillaQubit(reg, i)] = qubit_index
+                qubit_index += 1
+        else:
+            mqt_computation.add_qubit_register(size, name)
+            for i in range(size):
+                qubit_map[Qubit(reg, i)] = qubit_index
+                qubit_index += 1
+
+    clbit_index = 0
+    clbit_map = {}
+    qiskit_cregs = qiskit_circuit.cregs
+    for reg in qiskit_cregs:
+        size = reg.size
+        name = reg.name
+        mqt_computation.add_classical_bit_register(size, name)
+        for i in range(size):
+            clbit_map[Clbit(reg, i)] = clbit_index
+            clbit_index += 1
+
+    mqt_computation.gphase = qiskit_circuit.global_phase
+
+    data = qiskit_circuit.data
+    for inst in data:
+        instruction = inst[0]
+        qargs = inst[1]
+        cargs = inst[2]
+        params = instruction.params
+
+        _emplace_operation(mqt_computation, instruction, qargs, cargs, params, qubit_map, clbit_map)
+
+    # import initial layout and output permutation in case it is available
+    if qiskit_circuit.layout is not None:
+        _import_layouts(mqt_computation, qiskit_circuit)
+
+    mqt_computation.initialize_io_mapping()
+    return mqt_computation
+
+
+def _emplace_operation(
+    mqt_computation: QuantumComputation,
+    instr: Instruction,
+    qargs: tuple[Qubit] | list[Qubit],
+    cargs: tuple[Clbit] | list[Clbit],
```

Comment:
Iterable doesn't really work since I need to index stuff which is not an intrinsic property of an Iterable. It seems odd, but there doesn't seem to be a satisfactory solution except manually defining something like an "Indexable".

Reply:
There is. It is called `Sequence` ;-) My suggestion of `Iterable` was a little bit too generic.

Reply:
Ah, seems I googled badly. Changed it.